portfolio/src/main/webapp/scripts/featured_works.js

```diff
@@ -0,0 +1,143 @@
+const parentDir = ''; // TODO: add link
+
+fetch(parentDir + 'projects/projects.json')
+    .then(response => response.json())
+    .then(projectsJson => setupWorkSection(projectsJson))
+    .catch(err => console.error(err));
+
+/** 
+ * Sets up featured works section by creating project elements, and
+ * left- and right- buttons for navigation.
+ */
+function setupWorkSection(projectsJson) {
+  const projectsHtml = setupProjects(projectsJson);
+  const worksHtml = addNavButtons(projectsHtml);
+
+  const projectsPlaceholder = document.getElementById('projects');
+  ensureNonNull(projectsPlaceholder);
+  projectsPlaceholder.innerHTML = worksHtml;
+
+  displayFirstProj();
+  setupProjectNav();
+}
+
+/** Creates string which is used to display all projects. */
+function setupProjects(projectsJson) {
+  const projects = toProjects(projectsJson);
+  const projectDivs = projects.map(project => project.toDiv);
+  const projectsHtml = projectDivs.map(projectDiv => projectDiv.outerHTML)
+                          .join('');
+  return projectsHtml;
+}
+
+/** 
+ * Converts an array of projects that is represented as JSON objects to Projects. 
+ * @return {Array<Project>}
+ */
+const toProjects = (projectsJson) => {
+  return projectsJson.map(projectJson => toProject(projectJson));
```

Comment:
nit: You can simply do `return projectsJson.map(toProject)` here. Or remove this method, as it's self-explaining enough.